### ISSUE TITLE
S3+ Potential Security Vulnerability

### ISSUE THREAD

**Issue by tijohnson** [hyperledger/sawtooth-next-directory#226]
A dependency defined in requirements.dev.txt has known security vulnerabilities and should be updated.

Vulnerability found in cryto 2.6.1: https://nvd.nist.gov/vuln/detail/CVE-2018-6594

**Comment by agoldingay** [comment]
Chris has opened up a helpdesk ticket with the Hyperledger to address.

**Comment by jbobo** [comment]
@tijohnson where is this dependency? I don't see It in requirements.dev.txt or find anything when running `ack cryto` in the root project directory.

**Comment by jbobo** [comment]
nvm, @adamgering filled me in.

The dependency that uses cryto 2.6.1 is pycrypto, I used `ack` to get the files it shows up in:

```
sawtooth/lib/python3.7/site-packages/Crypto/Util/randpool.py
36:    See http://www.pycrypto.org/randpool-broken
39:        warnings.warn("This application uses RandomPool, which is BROKEN in older releases.  See http://www.pycrypto.org/randpool-broken",

sawtooth/lib/python3.7/site-packages/pycrypto-2.6.1.dist-info/RECORD
257:pycrypto-2.6.1.dist-info/INSTALLER,sha256=zuuue4knoyJ-UwPPXg8fezS7VCrXJQrAP7zeNuwvFQg,4
258:pycrypto-2.6.1.dist-info/METADATA,sha256=1l-9AVgkccANEeeXZ29e95C1BWovYM_VeUb1T4Lat0g,656
259:pycrypto-2.6.1.dist-info/RECORD,,
260:pycrypto-2.6.1.dist-info/WHEEL,sha256=_k7-jlcomYi2KWQV8j4dnQP2hhlJPv6RZQa_A7vSCGc,110
261:pycrypto-2.6.1.dist-info/top_level.txt,sha256=-W2wTtkxc1QnPUPRqBZ0bMwrhD8xRD13HIobFX-wDOs,7

sawtooth/lib/python3.7/site-packages/pycrypto-2.6.1.dist-info/METADATA
2:Name: pycrypto
5:Home-page: http://www.pycrypto.org/

sawtooth/lib/python3.7/site-packages/pipreqs/mapping
12:Crypto:pycrypto

requirements.dev.txt
10:pycrypto==2.6.1

shell/Dockerfile
36:    pycrypto \

tests/Dockerfile
35:    pycrypto \

server/Dockerfile
32:        pycrypto \

server/Dockerfile-dev
35:    pycrypto
```

**Comment by tijohnson** [comment]
You are correct the dependency on crypto 2.6.1 is through pycrypto. This is what the scan is looking at: https://github.com/hyperledger/sawtooth-next-directory/network/dependencies

**Comment by jbobo** [comment]
@tijohnson since it looks like all available versions of PyCrypto are vulnerable, and https://pypi.org/project/pycrypto/#history says the current version was released back in 2013, should we just find a suitable replacement package and refactor?

**Comment by jbobo** [comment]
No prob, thanks @tijohnson !

Actually, it looks like pipreqs mapped it from Crypto, and running `ack Crypto` shows:

```
transaction_creation/rbac_transaction_creation/common.py
23:from sawtooth_signing import CryptoFactory
43:    factory = CryptoFactory(sawtooth_signing.create_context("secp256k1"))

server/api/utils.py
22:from Crypto.Cipher import AES
(sawtooth) tm0493806:sawtooth-next-directory bobonana$
```

so it looks like the only place it's being used is server/api/utils.py:22 for an AES cipher. I'll look up a suitable replacement and see if we can just drop it in.

**Comment by kthblmfld** [comment]
So I see it in Python3.7, but I am wondering why it needs to be explicitly supplied in addition.

**Comment by chrisspanton** [comment]
Closed with PR #259
